### PR TITLE
Update homepage example for v1.4.0

### PR DESCRIPTION
**Pull request by dangra**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -47,9 +47,9 @@
         for title in response.css('h2.entry-title'):
             yield {'title': title.css('a ::text').extract_first()}
```
Comment (dangra) [inline index.html]: I was reluctant on using `.get()` here because it requires latest parsel too. What do you think?

File: index.html [hunk 2]
```diff
@@ -47,9 +47,9 @@
...
 
-        next_page = response.css('div.prev-post > a ::attr(href)').extract_first()
+        next_page = response.css('div.prev-post > a')
         if next_page:
-            yield scrapy.Request(response.urljoin(next_page), callback=self.parse)
+            yield response.follow(next_page[0], self.parse)
```
Comment (redapple) [inline index.html]: is the use of `next_page[0]` to show that `.follow()` allows passing a `Selector`?
I'm not sure I like this better than using `.extract_first()` and then `response.follow(next_page)`

File: index.html (hunk 1 above)
Comment (redapple) [inline index.html]: Yeah, `.get()` is still pretty young.

File: index.html (hunk 2 above)
Comment (redapple) [inline index.html]: we could also use a for loop and break after yield

Comment (dangra) [inline index.html]: Yes, an alternative is to change next_age code to:

```python

for next_page in response.css('div.prev-post > a'):
    yield response.follow(next_page, self.parse)
```

Comment (dangra) [inline index.html]: I must admit the example on how to get an attribute using CSS is compelling too.

File: index.html (hunk 1 above)
Comment (kmike) [inline index.html]: I think we can start using it in next release, and update Scrapy docs to use it as well. To do so we may increase parsel version in install_requires.

File: index.html (hunk 2 above)
Comment (dangra) [inline index.html]: ok, I just read your comment about "loop and yield", I will update the code and ignore the "break" suggestion :)

File: index.html (hunk 1 above)
Comment (dangra) [inline index.html]: sgtm

**Comment by dangra** [comment]
@redapple @kmike is it good to merge now?

File: index.html [hunk 3]
```diff
@@ -47,9 +47,8 @@
         for title in response.css('h2.entry-title'):
             yield {'title': title.css('a ::text').extract_first()}
 
-        next_page = response.css('div.prev-post > a ::attr(href)').extract_first()
-        if next_page:
-            yield scrapy.Request(response.urljoin(next_page), callback=self.parse)
+        for next_page in response.css('div.prev-post > a'):
+            yield response.follow(next_page, self.parse)
```
Comment (kmike) [inline index.html]: It looks fine to me, this is the approach I used for Scrapy tutorial.

Probably unrelated, but there is a small difference between old example and new example: if you write

```py
for next_page in response.css('div.prev-post > a ::attr(href)'):
    yield response.follow(next_page, self.parse)
```

(or the whole .extract_first(), etc. as before) `<a>` elements without href are skipped, while with 

```py
for next_page in response.css('div.prev-post > a'):
    yield response.follow(next_page, self.parse)
```

they raise an exception: `<a>` element is already selected, response.follow can't filter it out. Maybe adding a note to Scrapy docs makes sense: using `a` selectors directly is a bit easier and works fine in most case, but it is not 100% the same as using `a::attr(href)`.

Comment (dangra) [inline index.html]: the idea is to show a working example, no need to filter by href presence IMO.

Comment (dangra) [inline index.html]: by the way, +1 to adding a note in scrapy docs about `a::attr(href)` vs `a`

Comment (dangra) [inline index.html]: @redapple only you left to approve this PR. feel free to merge if you approve it.

**Pull request merged**